ruff.toml

```diff
@@ -0,0 +1,85 @@
+target-version = "py39"
+
+include = [
+    "package/kedro_viz/*.py",
+    "package/tests/*.py",
+    "package/features/*.py",
+    "demo-project/*.py",
+]
+
+[lint]
+select = [
+    "I", # Isort
+    "B", # Bugbear
+    "BLE", # Blind exceptions
+    "PL", # Pylint
+    "C90", # Mccabe complexity
+    "E", # Pycodestyle errors
+    "F", # Pyflakes
+    "W", # Pycodestyle warnings
+    "N", # PEP8-compliant object names
+    "SLF", # Private members access
+    "D101", # Class docstrings
+    "D103", # Function docstrings
+    "ARG", # Unused arguments
+    "T10", # Debug statements
+]
+ignore = [
+    "E203",
+    "E231",
+    "E266",
+    "E501",
+    "F401",
+    "B030", # Except handler is something other than exception class
+    "C405", # Inconsistent definition of literal collections
+    "PLR2004", # Magic values in comparisons
+]
+
+[lint.per-file-ignores]
+"*/cli_steps.py" = ["B011"] # assert False instead of AssertionError
+"*/base_deployer.py" = ["B024"] # ABCs without abstract methods
+"package/kedro_viz/__init__.py" = ["B028"] # Risky usage of positional arguments
```

Comment:
Q: does it make sense to ignore some of these specific issues in the code with noqa inline instead of here? Can keep the ignores that apply to multiple files here.

Reply:
I definitely wouldn't want to ignore the 'broad exception' rule in general, because I think we should avoid it as much as possible. I guess some others like # noqa: ARG002 should be fine.

Reply:
@rashidakanchwala 

_Broad exception_ rule has `BLE` code in `ruff` and is selected for entire repo ✅  (See line 14 of this file)

Reply:
@ankatiyar 

Good question!

The way I implemented it is as follows:

1. Whatever has an inline NOQA before that's applied to 1-2 places, stays as inline NOQA.
2. Whatever was a file-level `pylint: disable` statement at the beginning of the file is moved to `ruff.toml` from that separate file.

Let me explain the rationale for two.

Consider two implementations, using `package/kedro_viz/integrations/kedro/sqlite_store.py` as an example:

**Before**

```python
# pylint: broad-exception-caught
```

At the beginning of that particular file.

**After**

```toml
"package/kedro_viz/integrations/kedro/sqlite_store.py" = ["BLE"]
```

In `ruff.toml`

----

- In both cases, we essentially store the same directive - disable a particular rule in a particular file.
- However (2) makes it more explicit and centralizes all those directives, which would otherwise be scattered across tens of files, in a **single file**.
- Long term, that helps gradually make the codebase adhere to those rulesets, and delete those `per-file-ignores`. As opposed to distributed inline NOQAs that typically just never addressed.